### PR TITLE
Unified transfer function editor

### PR DESCRIPTION
**Pull request by cquammen**
Refactoring of the transfer function editor into a single chart and QVTKWidget.

Includes a fix for aligning the color bar with the left axis of the histogram chart.

### CODE REVIEW

**Comment by cquammen** [comment]
@cryos Please review at your convenience.

**Comment by cquammen** [comment]
Note alignment of color bar with histogram left side.

image

File: tomviz/CentralWidget.ui [hunk 1]
```diff
@@ -96,15 +68,15 @@
  </widget>
  <customwidgets>
   <customwidget>
-   <class>pqTabbedMultiViewWidget</class>
+   <class>QVTKWidget</class>
    <extends>QWidget</extends>
-   <header>pqTabbedMultiViewWidget.h</header>
+   <header location="global">qvtkwidget.h</header>
```
Comment (mathturtle) [inline tomviz/CentralWidget.ui]: Should be 'QVTKWidget.h'.  Case matters.  OSX's filesystem is a bit more forgiving about this sort of thing.  It was a build error on my Ubuntu machine.

Comment (cquammen) [inline tomviz/CentralWidget.ui]: Ah, thanks for catching. Designer must have done this. Actually, I don't know why the pqTabbedMultiViewWidget was changed at all. Let me revisit, but thanks for pointing this out.

File: tomviz/vtkChartHistogramColorOpacityEditor.cxx [hunk 2]
```diff
@@ -0,0 +1,262 @@
...
+    float bottomAxisHeight = this->GetHistogramAxis(vtkAxis::BOTTOM)->
+      GetBoundingRect(painter).GetHeight();
+    float verticalMargin = bottomAxisHeight;
+    y += colorBarThickness + verticalMargin - 5;
+    vtkRectf histogramChart(x, y, plotWidth, sceneHeight - y - this->Borders[vtkAxis::TOP]);
+    this->HistogramChart->SetSize(histogramChart);
+  }
+
+  return this->Superclass::Paint(painter);
+}
```
Comment (cryos) [inline tomviz/vtkChartHistogramColorOpacityEditor.cxx]: Not sure what editor you use, but try to ensure there is a newline at the end of files. I know mine just does it for me (vim or Qt Creator) but others don't seem to have this set as a default at least.

Comment (cquammen) [inline tomviz/vtkChartHistogramColorOpacityEditor.cxx]: Thanks for catching. I'm trying out Sublime Text 3 and newline insertion is off by default. Fixed now.

File: tomviz/CentralWidget.ui (hunk 1 above)
Comment (mathturtle) [inline tomviz/CentralWidget.ui]: Designer does some really strange things sometimes.  If you ever edit the MainWindow.ui file, there is a two column table in there that you have to set the number of columns back to 2.  Designer is convinced it should be 1 for some reason.

Comment (cquammen) [inline tomviz/CentralWidget.ui]: Done.

File: tomviz/vtkChartHistogramColorOpacityEditor.cxx (hunk 2 above)
Comment (cquammen) [inline tomviz/vtkChartHistogramColorOpacityEditor.cxx]: Done. I checked all the files in this branch's commits and verified they had newlines at the end of the file.

**Comment by cryos** [comment]
This looks great, the initial view looks much better, I think the interaction is working really well now. Merging.

**Pull request merged**